python_modules/libraries/dagster-datadog/dagster_datadog/resources.py

```diff
@@ -136,4 +141,6 @@ def dd_job():
             )
 
     """
-    return DatadogResource.from_resource_context(context).get_client()
+    resource = DatadogResource.from_resource_context(context)
+    resource.pre_execute(context)
+    return resource
```

Comment:
@benpankow This PR feels pretty good except for this. For the backwards compat case the user has to handle lifecycle hooks, which doesn't feel great.